packages/assets-controllers/src/NftController.ts

```diff
@@ -639,26 +830,25 @@ export class NftController extends BaseControllerV1<NftConfig, NftState> {
    * @param source - Whether the NFT was detected, added manually or suggested by a dapp.
    * @returns Promise resolving to the current NFT list.
    */
-  private async addIndividualNft(
+  async #addIndividualNft(
     tokenAddress: string,
     tokenId: string,
     nftMetadata: NftMetadata,
     nftContract: NftContract,
     chainId: Hex,
     userAddress: string,
     source: Source,
-  ): Promise<Nft[]> {
-    // TODO: Remove unused return
-    const releaseLock = await this.mutex.acquire();
+  ): Promise<void> {
+    const releaseLock = await this.#mutex.acquire();
     try {
-      tokenAddress = toChecksumHexAddress(tokenAddress);
+      const checksumHexAddress = toChecksumHexAddress(tokenAddress);
       const { allNfts } = this.state;
 
-      const nfts = allNfts[userAddress]?.[chainId] || [];
+      const nfts = [...(allNfts[userAddress]?.[chainId] || [])];
```

Comment:
nit: prefer nullish coalescing

```suggestion
      const nfts = [...(allNfts[userAddress]?.[chainId] ?? [])];
```